.editorconfig

```diff
@@ -1,134 +1,206 @@
+
+# top-most .editorconfig file
 root = true
 
-# Rules in this file were initially inferred by Visual Studio IntelliCode from the S:\Repos\dep.controls codebase based on best match to current usage at 11/15/2018
-# You can modify the rules from these initially generated values to suit your own policies
-# You can learn more about editorconfig here: https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference
-[*.cs]
+# defaults for all files
+[*]
+charset = utf-8
+end_of_line = crlf
+indent_size = 4
+indent_style = space
+insert_final_newline = true
+trim_trailing_whitespace = true
 
-#Core editorconfig formatting - indentation
+# markdown, diff overrides
+# two trailing spaces are required for <br/> and hard line-breaks in markdown files
+# see: (https://daringfireball.net/projects/markdown/syntax#p) and (http://spec.commonmark.org/0.27/#hard-line-break)
+[*.{md,diff}]
+trim_trailing_whitespace = false
 
-#use soft tabs (spaces) for indentation
-indent_style = space
-indent_size = 4
+[*.{md,xml,gprops,man,natvis}]
+indent_size = 2
+
+# manifest validation tool requires BOM
+[*.man]
 charset = utf-8-bom
-end_of_line = crlf
 
-#Formatting - indentation options
+# XML-based MSBuild and Visual Studio files
+[*.{props,targets,settings,*proj,vcxitems,filters}]
+indent_size = 2
 
-#indent switch case contents.
-csharp_indent_case_contents = true
-#csharp_indent_case_contents_when_block
-csharp_indent_case_contents_when_block = true
-#indent switch labels
-csharp_indent_switch_labels = true
+# Exceptions to the above *proj wildcard
+[*.vdproj]
+indent_size = 4
 
-#Formatting - new line options
+# Visual Studio uses hard tabs for SLN files, so don't fight it
+[*.sln]
+indent_style = tab
 
-#place catch statements on a new line
-csharp_new_line_before_catch = true
-#place else statements on a new line
-csharp_new_line_before_else = true
-#require braces to be on a new line for accessors, methods, object_collection, control_blocks, types, properties, and lambdas (also known as "Allman" style)
-csharp_new_line_before_open_brace = accessors, methods, object_collection, control_blocks, types, properties, lambdas
+# YAML overrides
+[*.{yml,yaml}]
+indent_size = 2
 
-#Formatting - organize using options
+# package.json overrides
+# Updating package.json with NPM will revert indentation to 2 spaces so to
+# reduce churn, let's align to NPM and specify indent size 2
+[package.json]
+indent_size = 2
 
-#do not place System.* using directives before other using directives
-dotnet_sort_system_directives_first = false
+# JSON formats
+[{*.json,*.testlist,testmd.definition}]
+indent_size = 2
 
-#Formatting - spacing options
+[*.{rc}]
+trim_trailing_whitespace = true
+insert_final_newline = true
 
-#require NO space between a cast and the value
+[*.{c,w}]
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.{c++,cc,cpp,cppm,cxx,h,h++,hh,hpp,hxx,inl,ipp,ixx,tlh,tli,idl}]
```

Comment:
Can *.ps1 and *.psm1 be handled here too?

Reply:
yes, but in this change, I want to copy os.2020. if you want that please start with a PR against that.